src/libasr/runtime/lfortran_intrinsics.c

```diff
@@ -623,6 +623,15 @@ LFORTRAN_API void _lfortran_strcat(char** s1, char** s2, char** dest)
     *dest = &(dest_char[0]);
 }
 
+// strcpy -----------------------------------------------------------
+
+LFORTRAN_API void _lfortran_strcpy(char** x, char *y)
+{
+    if (*x) free((void *)*x);
+    *x = (char*) malloc((strlen(y) + 1) * sizeof(char));
+    strcpy(*x, y);
```

Comment:
We are de-referencing `x` in all the three lines? So why not `_lfortran_strcpy(char* x, char *y)`? Any particular reasons?

Reply:
We are actually changing the pointer values and so that won't work as the pointer will be a "copy" variable. Notice the difference here:
```c
#include <stdio.h>

void fun(int *a) {
    int x = 10;
    a = &x;
}

void fun2(int **a) {
    int x = 10;
    *a = &x;
}

int main() {
    int b = 5;
    int *a = &b;
    fun(a);
    printf("%d\n", *a); // Still prints 5
    fun2(&a);
    printf("%d\n", *a); // prints 10
    return 0;
}
```

Reply:
I see. Was expecting this. Thanks for the confirmation. Going to approve. Please rebase on top of latest `main` and enable auto merge.